### PR TITLE
Tooling refactor preparing for configuration workflows

### PR DESCRIPTION
**Pull request by connorwstein**
## Motivation 
Various refactoring tasks prior to adding more configuration workflows. 
## Solution
- Incorporating ramp/changset rename
- Extracting chain contract deployment and state representation per  chain

### CODE REVIEW

File: integration-tests/deployment/address_book.go [hunk 1]
```diff
@@ -150,3 +150,9 @@ func NewMemoryAddressBook() *AddressBookMap {
 		AddressesByChain: make(map[uint64]map[string]TypeAndVersion),
 	}
 }
+
+func NewMemoryAddressBookFromMap(addressesByChain map[uint64]map[string]TypeAndVersion) *AddressBookMap {
```
Comment (connorwstein) [inline integration-tests/deployment/address_book.go:154]: helps with testing

File: integration-tests/deployment/address_book_test.go [hunk 2]
```diff
@@ -58,36 +56,57 @@ func TestAddressBook_Save(t *testing.T) {
...
-	ab2 := NewMemoryAddressBook()
-	require.NoError(t, ab2.Save(chainsel.TEST_90000003.Selector, addr3, onRamp100))
-	require.NoError(t, ab.Merge(ab2))
-	// Other address book should remain unchanged.
-	addresses, err = ab2.Addresses()
-	require.NoError(t, err)
-	assert.DeepEqual(t, addresses, map[uint64]map[string]TypeAndVersion{
-		chainsel.TEST_90000003.Selector: {
-			addr3: onRamp100,
+func TestAddressBook_Merge(t *testing.T) {
```
Comment (connorwstein) [inline integration-tests/deployment/address_book_test.go:61]: explicit merge test

File: integration-tests/deployment/ccip/add_lane.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+	"github.com/smartcontractkit/chainlink/integration-tests/deployment"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/offramp"
+
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/onramp"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/price_registry"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/router"
+)
+
+func AddLane(e deployment.Environment, state CCIPOnChainState, from, to uint64) error {
```
Comment (connorwstein) [inline integration-tests/deployment/ccip/add_lane.go:17]: separate file as we'll add an explicit add_lane_test.go soon

File: integration-tests/deployment/ccip/changeset/2_initial_deploy_test.go [hunk 4]
```diff
@@ -27,76 +25,26 @@ import (
...
-	// First chain is home chain.
-	for chainSel := range chains {
-		homeChainEVM, _ = chainsel.ChainIdFromSelector(chainSel)
-		homeChainSel = chainSel
-		break
-	}
-	ab, err := ccipdeployment.DeployCapReg(lggr, chains, homeChainSel)
-	require.NoError(t, err)
+	ctx := ccipdeployment.Context(t)
+	tenv := ccipdeployment.NewDeployedTestEnvironment(t, lggr)
```
Comment (connorwstein) [inline integration-tests/deployment/ccip/changeset/2_initial_deploy_test.go:31]: extracted a helper which loads a memory env with cap reg already set up (and nodes pointed at it)

File: integration-tests/deployment/ccip/deploy.go [hunk 5]
```diff
@@ -114,308 +113,58 @@ type DeployCCIPContractConfig struct {
...
-					chain.DeployerKey,
-					chain.Client,
-					false,
-				)
-				return ContractDeploy[*maybe_revert_message_receiver.MaybeRevertMessageReceiver]{
-					receiverAddr, receiver, tx, deployment.NewTypeAndVersion(CCIPReceiver, deployment.Version1_0_0), err2,
-				}
-			})
+	for _, chain := range e.Chains {
+		ab, err = DeployChainContracts(e, chain, ab)
```
Comment (connorwstein) [inline integration-tests/deployment/ccip/deploy.go:142]: extracted a helper here which will be useful for the add new chain workflow

File: integration-tests/deployment/ccip/state.go [hunk 6]
```diff
@@ -26,34 +26,38 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/weth9"
 )
 
+type CCIPChainState struct {
```
Comment (connorwstein) [inline integration-tests/deployment/ccip/state.go:29]: pulled out a per chain struct, also will be useful for the add chain workflow

File: integration-tests/deployment/ccip/state.go [hunk 7]
```diff
@@ -137,134 +146,130 @@ func (s CCIPOnChainState) Snapshot(chains []uint64) (CCIPSnapShot, error) {
...
-	state, err := GenerateOnchainState(e, ab)
+	state, err := LoadOnchainState(e, ab)
 	if err != nil {
 		return CCIPSnapShot{}, err
 	}
 	return state.Snapshot(e.AllChainSelectors())
 }
 
-func GenerateOnchainState(e deployment.Environment, ab deployment.AddressBook) (CCIPOnChainState, error) {
+func LoadOnchainState(e deployment.Environment, ab deployment.AddressBook) (CCIPOnChainState, error) {
```
Comment (connorwstein) [inline integration-tests/deployment/ccip/state.go:156]: I think load is more accurate

File: integration-tests/deployment/ccip/changeset/1_cap_reg.go [hunk 8]
```diff
@@ -0,0 +1,21 @@
+package changeset
+
+import (
+	"github.com/smartcontractkit/chainlink/integration-tests/deployment"
+	ccipdeployment "github.com/smartcontractkit/chainlink/integration-tests/deployment/ccip"
+)
+
+// Separate migration because cap reg is an env var for CL nodes.
+func Apply0001(env deployment.Environment, homeChainSel uint64) (deployment.ChangeSetOutput, error) {
```
Comment (asoliman92) [inline integration-tests/deployment/ccip/changeset/1_cap_reg.go:9]: This is not used anywhere here now. Is it expected?

**Review by asoliman92 — APPROVED** [review]
Only question about unused `Apply0001`, otherwise looks good.

File: integration-tests/deployment/ccip/test_helpers.go [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/chainlink/integration-tests/deployment"
+	"github.com/smartcontractkit/chainlink/integration-tests/deployment/memory"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+)
+
+// Context returns a context with the test's deadline, if available.
+func Context(tb testing.TB) context.Context {
```
Comment (AnieeG) [inline integration-tests/deployment/ccip/test_helpers.go:18]: can be replaced by a call to [testcontext.go](https://github.com/smartcontractkit/chainlink-testing-framework/blob/main/utils/testcontext/testcontext.go#L9)

File: integration-tests/deployment/ccip/add_lane.go [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+	// TODO: Batch
+	tx, err := state.Chains[from].Router.ApplyRampUpdates(e.Chains[from].DeployerKey, []router.RouterOnRamp{
+		{
+			DestChainSelector: to,
+			OnRamp:            state.Chains[from].EvmOnRampV160.Address(),
+		},
+	}, []router.RouterOffRamp{}, []router.RouterOffRamp{})
```
Comment (AnieeG) [inline integration-tests/deployment/ccip/add_lane.go:24]: missing errcheck

File: integration-tests/deployment/ccip/add_lane.go [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+	if err := deployment.ConfirmIfNoError(e.Chains[from], tx, err); err != nil {
+		return err
+	}
+	tx, err = state.Chains[from].EvmOnRampV160.ApplyDestChainConfigUpdates(e.Chains[from].DeployerKey,
```
Comment (AnieeG) [inline integration-tests/deployment/ccip/add_lane.go:28]: missing errcheck

File: integration-tests/deployment/ccip/add_lane.go [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+					SourceToken: state.Chains[from].Weth9.Address(),
+					UsdPerToken: deployment.E18Mult(4000),
+				},
+			},
+			GasPriceUpdates: []price_registry.InternalGasPriceUpdate{
+				{
+					DestChainSelector: to,
+					UsdPerUnitGas:     big.NewInt(2e12),
+				},
+			}})
```
Comment (AnieeG) [inline integration-tests/deployment/ccip/add_lane.go:56]: missing errcheck

File: integration-tests/deployment/ccip/add_lane.go [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+
+	tx, err = state.Chains[to].EvmOffRampV160.ApplySourceChainConfigUpdates(e.Chains[to].DeployerKey,
+		[]offramp.OffRampSourceChainConfigArgs{
+			{
+				Router:              state.Chains[to].Router.Address(),
+				SourceChainSelector: from,
+				IsEnabled:           true,
+				OnRamp:              common.LeftPadBytes(state.Chains[from].EvmOnRampV160.Address().Bytes(), 32),
+			},
+		})
```
Comment (AnieeG) [inline integration-tests/deployment/ccip/add_lane.go:81]: missing errcheck

File: integration-tests/deployment/ccip/add_lane.go [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+	if err := deployment.ConfirmIfNoError(e.Chains[to], tx, err); err != nil {
+		return err
+	}
+	tx, err = state.Chains[to].Router.ApplyRampUpdates(e.Chains[to].DeployerKey, []router.RouterOnRamp{}, []router.RouterOffRamp{}, []router.RouterOffRamp{
+		{
+			SourceChainSelector: from,
+			OffRamp:             state.Chains[to].EvmOffRampV160.Address(),
+		},
+	})
```
Comment (AnieeG) [inline integration-tests/deployment/ccip/add_lane.go:90]: missing errcheck

File: integration-tests/deployment/ccip/add_lane.go [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+	return deployment.ConfirmIfNoError(e.Chains[to], tx, err)
+}
+
+func defaultPriceRegistryDestChainConfig() price_registry.PriceRegistryDestChainConfig {
```
Comment (AnieeG) [inline integration-tests/deployment/ccip/add_lane.go:94]: might help to have all defaults in a single place

**Review by AnieeG — COMMENTED** [review]
partial review

File: integration-tests/deployment/ccip/changeset/1_cap_reg.go (hunk 8 above)
Comment (connorwstein) [reply integration-tests/deployment/ccip/changeset/1_cap_reg.go:9]: Yeah will be imported/used in the deployments repo

**Review by AnieeG — APPROVED** [review]

File: integration-tests/deployment/ccip/add_lane.go (hunk 15 above)
Comment (connorwstein) [reply integration-tests/deployment/ccip/add_lane.go:94]: these are kinda lane specific so in add_lane seems ok

**Pull request merged**